### ISSUE TITLE
False-negative `override` check when overriding `staticmethod` with non-static method

### ISSUE THREAD

**Issue by bersbersbers** [microsoft/pyright#7392]
**Describe the bug**
By mistake(?), I produced the below code.

`mypy` reports both `overrides`:
```
bug.py:12: error: Signature of "method" incompatible with supertype "Super"  [override]
bug.py:12: note:      Superclass:
bug.py:12: note:          @staticmethod
bug.py:12: note:          def method() -> None
bug.py:12: note:      Subclass:
bug.py:12: note:          def method(self) -> None
bug.py:24: error: Signature of "method" incompatible with supertype "SuperProtocol"  [override]
bug.py:24: note:      Superclass:
bug.py:24: note:          @staticmethod
bug.py:24: note:          def method() -> None
bug.py:24: note:      Subclass:
bug.py:24: note:          def method(self) -> None
Found 2 errors in 1 file (checked 1 source file)
```

`pyright` reports neither. One of two must be wrong - who is it? :)

Or should I be allowed to `override` a `staticmethod` with a regular method? (Why would anyone do that? To avoid `ruff`'s [PLR6301](https://docs.astral.sh/ruff/rules/no-self-use/) in the super class, for instance.)

**Code or Screenshots**
```python
from typing import Protocol, override


class Super:
    @staticmethod
    def method() -> None:
        print("Super")


class Derived(Super):
    @override
    def method(self) -> None:
        print("Derived")


class SuperProtocol(Protocol):
    @staticmethod
    def method() -> None:
        pass


class DerivedFromProtocol(SuperProtocol):
    @override
    def method(self) -> None:
        pass
```

**Comment by erictraut** [comment]
I'm not able to repro the problem. Pyright reports the same two errors as my. I'm guessing that you have either disabled the `reportIncompatibleMethdOverride` check or are running a really old version of pyright. Please update to the latest and verify your configuration settings.

**Comment by bersbersbers** [comment]
> you have either disabled the `reportIncompatibleMethdOverride` check

You are right - sorry for that! I do have it disabled since `pyright` checks the parameter names, and `PySide6` uses auto-generated stubs which look like this:

```python
def setEnabled(self, arg__1: bool) -> None: ...
```

and would like to name my parameters. 

I guess `PySide6` should prefer

```python
def setEnabled(self, arg__1: bool, /) -> None: ...
```

**Comment by bersbersbers** [comment]
fyi, I filed https://bugreports.qt.io/browse/PYSIDE-2624
